app/routes.py

```diff
@@ -1 +1,268 @@
+    }, 200
+
+    return {"task": task.to_dict()}, 200
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return {"task": task.to_dict()}, 200
+
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def task_mark_complete(task_id):
+    task = validate_model(Task, task_id) 
+
+    task.completed_at = date.today()
+
+    db.session.commit()
+
+    key = os.environ.get("key")
+    url_path = "https://slack.com/api/chat.postMessage"
+    body = {
+        "channel": "C056W45DVEH",
+        "text": f"Someone just completed the task {task.title}"
+    }
+
+    header = {"Authorization": f"Bearer {key}"}
+
+    slack = requests.post(url_path, headers=header, json=body)
+
+    return {"task": task.to_dict()}, 200
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def task_mark_incomplete(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return {"task": task.to_dict()}, 200
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return jsonify({"details":f'Task {task.task_id} "{task.title}" successfully deleted'}), 200
+
+# - - - - - - - - - - - - - - - - - #
+###~~~--- goal model routes ---~~~###
+# - - - - - - - - - - - - - - - - - #
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@goal_bp.route("", methods=['POST'])
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal(
+            title = request_body["title"])
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+@goal_bp.route("/<goal_id>/tasks", methods=['POST'])
+def assign_tasks_to_goal(goal_id):
+    request_body = request.get_json()
+    response_tasklist = request_body["task_ids"]
+
+    goal = validate_model(Goal, goal_id)
+
+    for task_id in response_tasklist:
+        single_task = validate_model(Task, task_id)
+        
+        if single_task:
+            goal.tasks.append(single_task)
+
+    db.session.commit()
+
+    return {
+        "id": goal.goal_id,
+        "task_ids": response_tasklist
+        }, 200
+
+@goal_bp.route("/<goal_id>/tasks", methods=['GET'])
+def read_goal_tasks(goal_id):
+    goal = validate_model(Goal, goal_id)
+    tasklist = []
+
+    for task in goal.tasks:
+        tasklist.append({
+            "id": task.task_id,
+            "goal_id": goal.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": True if task.completed_at else False
+        })
```

Comment:
Do you think that this functionality could be moved into a class method? If so, what would that look like?